File: CPP/TramMaxPassen.cpp
#include<iostream>
using namespace std;
int main(){
    int stops,ans = 0;
    int maxpassenger = INT32_MIN;
    cin>>stops;
    int train[stops*2];
    for(int i = 0 ; i < stops*2; i++){
        int passenger;
        cin>>passenger;
        train[i] = passenger;
    }
    for(int i = 0 ; i < stops*2; i++){
        if(i == 1){
            ans = train[0] + train[i];
        }
        else if (i>1)
        {
            ans = ans + train[i+1] - train[i];
            i++;
        }
        maxpassenger = max(ans , maxpassenger);
    }
    cout<<maxpassenger;
    return 0;
}

File: JAVA/TramMaxPassen.java
import java.util.ArrayList;
import java.util.Scanner;
/*
 * author : pandeyanubhav888
 * title : tram
 */
public class TramMaxPassen{
    public static void main(String[] args) {
        ArrayList<Integer> passen = new ArrayList<>();
        int maxpassenger = Integer.MIN_VALUE;
        Scanner sc = new Scanner(System.in);
        int stops = sc.nextInt();
        int ans = 0;
        for(int i = 0 ; i < stops*2 ; i++){
            int passenger = sc.nextInt();
            passen.add(passenger);
        }
        for(int i = 0 ; i < stops*2 ; i++){
            if(i == 1){
                ans = passen.get(0) + passen.get(i);
            }
            else if(i > 1){
                ans = ans + passen.get(i+1) - passen.get(i);
                i++;
            }
            maxpassenger = Math.max(ans, maxpassenger);
        }
        System.out.println(maxpassenger);
        sc.close();
    }
}